extensions/amp-story/0.1/media-pool.js

```diff
@@ -457,41 +554,54 @@ export class MediaPool {
    * @private
    */
   swapPoolMediaElementIntoDom_(domMediaEl, poolMediaEl, sources) {
-    const ampMediaForPoolEl = ampMediaElementFor(poolMediaEl);
-    const ampMediaForDomEl = ampMediaElementFor(domMediaEl);
     poolMediaEl[REPLACED_MEDIA_PROPERTY_NAME] = domMediaEl.id;
 
     return this.enqueueMediaElementTask_(poolMediaEl,
         new SwapIntoDomTask(domMediaEl, this.vsync_))
         .then(() => {
-          this.maybeResetAmpMedia_(ampMediaForPoolEl);
-          this.maybeResetAmpMedia_(ampMediaForDomEl);
           this.enqueueMediaElementTask_(poolMediaEl,
               new UpdateSourcesTask(sources, this.vsync_));
           this.enqueueMediaElementTask_(poolMediaEl, new LoadTask());
+
+          this.triggerAllocationEvent_(poolMediaEl, /* isAllocated */ true);
         }, () => {
           this.forceDeallocateMediaElement_(poolMediaEl);
         });
   }
 
-
   /**
-   * @param {?Element} componentEl
-   * @private
+   * @param {!Element} element
+   * @param {boolean} isAllocated
    */
-  maybeResetAmpMedia_(componentEl) {
-    if (!componentEl) {
-      return;
-    }
+  triggerAllocationEvent_(element, isAllocated) {
+    const win = this.win_;
+    const {parentNode} = element;
+    const type = isAllocated ?
+        MediaPoolEvents.ALLOCATED :
+        MediaPoolEvents.DEALLOCATED;
 
-    if (componentEl.tagName.toLowerCase() == 'amp-audio') {
-      // TODO(alanorozco): Implement reset for amp-audio
-      return;
-    }
+    const detailOptional = !isAllocated ? this.getDeallocationDetail_(element) :
+        null;
 
-    componentEl.getImpl().then(impl => impl.resetOnDomChange());
+    const event = createCustomEvent(win, type, detailOptional, {bubbles: true});
+
+    dev().assertElement(parentNode).dispatchEvent(event);
   }
 
+  /**
+   * Gets a snapshot of the video state on deallocation time. This is so that
+   * the media pool consumer can know details of the video without access to the
+   * element itself.
+   * @return {!MediaInfoDef}
+   */
+  getDeallocationDetail_(element) {
+    const {currentTime, duration, paused} = element;
+    return {
+      currentTime,
+      duration,
+      paused: true,
```

Comment:
Well, hopefully anyway 😉 

Should we let this actually just be `paused`, in case we have bugs?